### DISCUSSION TITLE
groupbar title background

### DISCUSSION THREAD

**Discussion by MathieuMoalic** [hyprwm/Hyprland#3284] [Q&A] [answered]
Hi there,

![image](https://github.com/hyprwm/Hyprland/assets/21007252/0317113a-708b-49eb-b235-752a2c925b40)

How does one add a background color to the groupbar title. I found how to change the color of the bottom border and add or remove a gradient of which I am not even sure I control the color.

Any ideas?

**Comment by MightyPlaza** [comment]
you have to restart hyprland

**Reply by MathieuMoalic** [reply]
Hum, restarting didn't change anything. I went throught all the configuration variables and I can't find a single one that sets the groupbar title color.

**Reply by MightyPlaza** [reply]
`groupbar_text_color`

**Reply by MathieuMoalic** [reply]
Sorry I made a mistake in my answer, I meant "add a **background** color to the groupbar title". The same way that i3 or sway would have **backgr.** :
| class | border  | **backgr.** | text | indicator | child_border | 
| --- | --- | --- | --- | --- |  --- | 
| client.focused          | #44475A | **#6272A4** | #F8F8F2 | #6272A4   | #6272A4 | 
| client.focused_inactive | #6272A4 | **#44475A** | #F8F8F2 | #44475A   | #44475A | 

such as:
![image](https://github.com/hyprwm/Hyprland/assets/21007252/90e06c0f-e828-41e8-95c7-16d87a3a0058)

**Reply by MightyPlaza** [reply, marked as answer]
unless you're using a gradient you can't (see https://github.com/hyprwm/Hyprland/pull/3197)

**Reply by MathieuMoalic** [reply]
Thanks for the answer, I was hoping it would be possible at the moment. I will wait until the feature is available.

**Reply by Kantis** [reply]
Not sure if new, but it's possible in Hyprland 0.49 using this config:

```
group {
  groupbar {
    height = 1
    font_size = 20
    # about half the indicator height
    text_offset = -17
    
    # Make the indicator tall enough to render text inside
    indicator_height = 34
  }
}
```

**Reply by onion108** [reply]
> Not sure if new, but it's possible in Hyprland 0.49 using this config:
> 
> ```
> group {
>   groupbar {
>     height = 1
>     font_size = 20
>     # about half the indicator height
>     text_offset = -17
>     
>     # Make the indicator tall enough to render text inside
>     indicator_height = 34
>   }
> }
> ```

nice hack! solved my problem

**Comment by Kantis** [comment]
Posting this as a top-level comment as well, in case someone else finds this thread when trying to get groupbar w/ background:

> Not sure if new, but it's possible in Hyprland 0.49 using this config:

```
group {
  groupbar {
    height = 1
    font_size = 20
    # about half the indicator height
    text_offset = -17
    
    # Make the indicator tall enough to render text inside
    indicator_height = 34
  }
}
```
